generate.py

```diff
@@ -80,7 +89,7 @@ def decode_one_token(
 ) -> Tuple[torch.Tensor, torch.Tensor]:
     # input_pos: [B, 1]
     assert input_pos.shape[-1] == 1
-    logits = model(x, input_pos)
+    logits = model(x, input_pos, mask=get_mask(x, input_pos, 128256))
```

Comment:
We need to write a `KVCacheGist` cache which is similar to `KVCacheFull` except that during prefill stage we only store the GIST tokens.

I can do this but this should be reverted.  We don't need mask (see below too)